index.js

```diff
@@ -44,12 +44,31 @@ var HOST_PORT_PATTERN = /^(\d+.\d+.\d+.\d+):\d+$/;
 var MAX_JOIN_DURATION = 300000;
 var PROXY_REQ_PROPS = ['keys', 'dest', 'req', 'res'];
 
+var ArgumentRequiredError = TypedError({
```

Comment:
suggestion: consider placing these errors in an errors.js file

This makes all the errors visible in a single file, might be more friendly.